examples/atomic-counters/atomic-counters.go

```diff
@@ -7,43 +7,40 @@
 
 package main
 
-import "fmt"
-import "time"
-import "sync/atomic"
+import (
+	"fmt"
+	"sync"
+	"sync/atomic"
+)
 
 func main() {
 
 	// We'll use an unsigned integer to represent our
 	// (always-positive) counter.
 	var ops uint64
 
-	// To simulate concurrent updates, we'll start 50
-	// goroutines that each increment the counter about
-	// once a millisecond.
+	// A WaitGroup will help us wait for all goroutines
+	// to finish their work.
+	var wg sync.WaitGroup
+
+	// We'll start 50 goroutines that each increment the
+	// counter exactly 1000 times.
 	for i := 0; i < 50; i++ {
+		wg.Add(1)
+
 		go func() {
-			for {
+			for c := 0; c < 1000; c++ {
 				// To atomically increment the counter we
 				// use `AddUint64`, giving it the memory
 				// address of our `ops` counter with the
 				// `&` syntax.
 				atomic.AddUint64(&ops, 1)
-
-				// Wait a bit between increments.
-				time.Sleep(time.Millisecond)
 			}
+			wg.Done()
 		}()
 	}
 
-	// Wait a second to allow some ops to accumulate.
-	time.Sleep(time.Second)
-
-	// In order to safely use the counter while it's still
-	// being updated by other goroutines, we extract a
-	// copy of the current value into `opsFinal` via
-	// `LoadUint64`. As above we need to give this
-	// function the memory address `&ops` from which to
-	// fetch the value.
-	opsFinal := atomic.LoadUint64(&ops)
-	fmt.Println("ops:", opsFinal)
+	// Wait until all the goroutines are done.
+	wg.Wait()
+	fmt.Println("ops:", ops)
```

Comment:
Do you think there's a risk people think this means it's safe to read something like `ops` directly like this under concurrent access?

Reply:
Good point! I'll add a comment explaining this, which incidentally will also mention `LoadUint64`